src/DGtal/geometry/surfaces/estimation/PlaneProbingTetrahedronEstimator.h

```diff
@@ -0,0 +1,332 @@
+
+namespace DGtal
+{
+    /**
+     * Probing mode for plane-probing estimators.
+     */
+    enum class ProbingMode
+    {
+        H,
+        R,
+        R1,
+    };
+
+    /**
+     * Display a mode on the standard output.
+     *
+     * @param aOs the output stream.
+     * @param aMode the mode to display.
+     */
+    inline
+    std::ostream& operator<< (std::ostream& aOs, ProbingMode const& aMode)
+    {
+        switch (aMode)
+        {
+            case ProbingMode::H:
+                aOs << "H";
+                break;
+
+            case ProbingMode::R:
+                aOs << "R";
+                break;
+
+            case ProbingMode::R1:
+                aOs << "R1";
+                break;
+        }
+
+        return aOs;
+    }
+
+  /////////////////////////////////////////////////////////////////////////////
+  // template class PlaneProbingTetrahedronEstimator
+  /**
+   * Description of template class 'PlaneProbingTetrahedronEstimator' <p>
+   * \brief Aim: A class that locally estimates a normal on a digital set using only
+   * a predicate "does a point x belong to the digital set or not?".
+   *
+   * @tparam TPredicate the probing predicate, a model of concepts::CPointPredicate.
+   * @tparam mode the probing mode, see DGtal::ProbingMode.
+   */
+  template <typename TPredicate, ProbingMode mode = ProbingMode::H>
+  class PlaneProbingTetrahedronEstimator
+  {
+    BOOST_CONCEPT_ASSERT((concepts::CPointPredicate<TPredicate>));
+
+    // ----------------------- Public types ------------------------------
+  public:
+      using Self            = PlaneProbingTetrahedronEstimator<TPredicate, mode>;
+      using Predicate       = TPredicate;
+      using Point           = typename Predicate::Point;
+      using Neighborhood    = PlaneProbingNeighborhood<TPredicate>;
+      using Triangle        = typename Neighborhood::Triangle;
+      using Integer         = typename Point::Coordinate;
+      using ProbingRay      = typename Neighborhood::ProbingRay;
+      using Permutation     = typename ProbingRay::Permutation;
+      using Quantity        = Point;
+      using HexagonState    = typename Neighborhood::HexagonState;
+      using UpdateOperation = typename Neighborhood::UpdateOperation;
+
+    // ----------------------- Standard services ------------------------------
+  public:
+    /**
+     * Default constructor.
+     */
+    PlaneProbingTetrahedronEstimator() = delete;
+
+    /**
+     * Constructs a plane probing tetrahedron estimator from an initial frame and a probing predicate.
+     *
+     * @param aPoint the base point of the initial frame.
+     * @param aM the three vectors of the initial frame.
+     * @param aPredicate the probing predicate.
+     */
+    PlaneProbingTetrahedronEstimator (Point const& aPoint, Triangle const& aM, Predicate const& aPredicate);
+
+    /**
+     * Destructor.
+     */
+    virtual ~PlaneProbingTetrahedronEstimator();
```

Comment:
Why is the destructor virtual ? I do not see a base class or virtual/abstract methods ?

Reply:
You're right, I removed the qualifier.